test/integration/targets/reboot/tasks/main.yml

```
@@ -1,39 +1,58 @@
-- name: Test reboot
-  when: ansible_facts.virtualization_type | default('') not in ['docker', 'container', 'containerd']
-  block:
-    # This block can be removed once we have a mechanism in ansible-test to separate
-    # the control node from the managed node.
-    - block:
-        - name: Write temp file for sanity checking this is not the controller
-          copy:
-            content: 'I am the control node'
-            dest: /tmp/Anything-Nutlike-Nuzzle-Plow-Overdue
-          delegate_to: localhost
-          connection: local
-          when: inventory_hostname == ansible_play_hosts[0]
+- name: Memorize whether we're in a containerized environment
+  set_fact:
+    in_container_env: >-
+      {{
+        (
+          (
+            ansible_facts.virtualization_type
+            | default('')
+          )
+          in ['docker', 'container', 'containerd']
+        )
+        | bool
+      }}
+
+- name: Check split state
+  stat:
+    path: "{{ output_dir }}"
+  register: split
+  ignore_errors: yes
 
-        - name: See if the temp file exists on the managed node
-          stat:
-            path: /tmp/Anything-Nutlike-Nuzzle-Plow-Overdue
-          register: controller_temp_file
+- name: Memorize whether we're in a split controller mode
+  set_fact:
+    in_split_controller_mode: >-
+      {{
+        (
+          split is not success
+          or not split.stat.exists
+        )
+        | bool
+      }}
```

Since the `set_fact` expressions are neither long or complex, they should each be on a single line:

```suggestion
- name: Set facts
  set_fact:
    in_split_controller_mode: "{{ (split is not success or not split.stat.exists) }}"
    in_container_env: "{{ (ansible_facts.virtualization_type in ['docker', 'container', 'containerd']) }}"
```

---

Oh, I've already merged the tasks locally. Regarding singlelining, it's not always about complexity. It's also about the screen to font-size ration that's an accessibility limitation for me. I hope you'll forgive me if I keep some bits multilined.